miden-lib/asm/miden/contracts/faucets/basic_fungible.masm

```diff
@@ -47,67 +48,75 @@ export.basic::auth_tx_rpo_falcon512
     push.METADATA_SLOT exec.account::get_item drop drop drop
-    # => [max_supply, amount, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [max_supply, amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
 
     # get total issuance of this faucet so far and add amount to be minted
     exec.faucet::get_total_issuance
-    # => [total_issuance, max_supply, amount, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [total_issuance, max_supply, amount, tag, aux, note_type, execution_hint, RECIPIENT, 
+    #     pad(7)]
 
     # compute maximum amount that can be minted, max_mint_amount = max_supply - total_issuance
     sub
-    # => [max_supply - total_issuance, amount, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [max_supply - total_issuance, amount, tag, aux, note_type, execution_hint, RECIPIENT, 
+    #     pad(7)]
 
     # check that amount =< max_supply - total_issuance, fails if otherwise
     dup.1 gte assert.err=ERR_FUNGIBLE_ASSET_DISTRIBUTE_WOULD_CAUSE_MAX_SUPPLY_TO_BE_EXCEEDED
-    # => [asset, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [amount, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
 
     # creating the asset
     exec.asset::create_fungible_asset
-    # => [ASSET, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [ASSET, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
 
     # mint the asset; this is needed to satisfy asset preservation logic.
     exec.faucet::mint
-    # => [ASSET, tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [ASSET, tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
 
     # store and drop the ASSET
     loc_storew.0 dropw
-    # => [tag, aux, note_type, execution_hint, RECIPIENT, ...]
+    # => [tag, aux, note_type, execution_hint, RECIPIENT, pad(7)]
 
     # create a note
     exec.tx::create_note
-    # => [note_idx, EMPTY_WORD, EMPTY_WORD, ...]
+    # => [note_idx, pad(15)]
 
     # load the ASSET and add it to the note
     movdn.4 loc_loadw.0 exec.tx::add_asset_to_note movup.4
-    # => [note_idx, ASSET, EMPTY_WORD, ...]
+    # => [note_idx, ASSET, pad(11)]
 end
 
 #! Burns fungible assets.
 #!
-#! Input: [ASSET]
-#! Outputs: []
+#! Inputs:  [ASSET, pad(12)]
+#! Outputs: [pad(16)]
 #!
+#! Where:
 #! - ASSET is the fungible asset to be burned.
 #!
-#! Fails if:
-#! - The transaction is executed against an account which is not a fungible asset faucet.
-#! - The transaction is executed against a faucet which is not the origin of the specified asset.
-#! - The amount about to be burned is greater than the outstanding supply of the asset.
+#! Panics if:
+#! - the transaction is executed against an account which is not a fungible asset faucet.
+#! - the transaction is executed against a faucet which is not the origin of the specified asset.
+#! - the amount about to be burned is greater than the outstanding supply of the asset.
+#!
+#! Annotations: call
```

Comment:
I'm wondering whether instead of "Annotations" we should just say "Invocation" here and in other similar places. Seems like for now this is the only type of annotations we have, and saying "invocation" would make the meaning of `call` a bit more clear.

Reply:
We also have `#! Annotations: exec` for the procedures in the `miden` lib, but probably it will be fine to use `Invocation` even with `exec`